File: plugins/xtest/examples/phpunit/moneytest.php
<?php
require_once 'PHPUnit/Framework.php';
require_once 'money.php';
 
class MoneyTest extends PHPUnit_Framework_TestCase {

    public function testCreate() {
	$m = new Money(5, "USD");
        $this->assertEquals(5, $m->amount());
	$this->assertEquals("USD", $m->currency());
    }
 
    /**
     * @expectedException Exception
     */
    public function testNegative() {
	$m = new Money(-5, "USD");
    }

    public function testZero() {
	$m = new Money(0, "USD");
	$this->assertEquals(0, $m->amount());
	$this->assertEquals("USD", $m->currency());
    }
}
?>

File: plugins/xtest/examples/junit3/MoneyTest.java
import junit.framework.Test;
import junit.framework.TestCase;
import junit.framework.TestSuite;
import java.lang.Exception;

public class MoneyTest extends TestCase {
    public void testCreate() throws Exception {
	Money m = new Money(5, "USD");
	assertEquals(5, m.amount());
	assertEquals("USD", m.currency());
    }

    public void testNegative() {
	try {
	    new Money(-5, "FOO");
	    fail("Should throw NegativeAmountException");
	} catch (Money.NegativeAmountException e) {}
    }

    public void testZero() throws Exception {
	Money m = new Money(0, "USD");
	assertEquals(0, m.amount());
    }

    public static Test suite() {
	return new TestSuite(MoneyTest.class);
    }
}
